lib/util/cleverMerge.js

```diff
@@ -82,56 +198,218 @@ const cachedSetProperty = (obj, property, value) => {
  * @returns {object} merged object of first and second object
  */
 const cleverMerge = (first, second) => {
-	let newObject = { ...first };
-	for (const key of Object.keys(second)) {
-		if (!(key in newObject)) {
-			newObject[key] = second[key];
-			continue;
+	if (second === undefined) return first;
+	if (first === undefined) return second;
+	const firstInfo = parseObject(first);
+	const secondInfo = parseObject(second);
```

Comment:
cache results at least for first one.. not sure..

Reply:
yep, that makes sense as these objects could be (and are) used multiple times. It also makes sense to cache merging of nested objects.